coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/functions/HiveFunctionResolver.java

```diff
@@ -182,24 +197,39 @@ public Collection<Function> tryResolveAsDaliFunction(String functionName, @Nonnu
     if (funcClassName == null) {
       return ImmutableList.of();
     }
-    final Collection<Function> Functions = registry.lookup(funcClassName);
-    if (Functions.size() == 0) {
+    // If the UDF class name is versioned, remove the versioning prefix, which allows user to
+    // register the unversioned UDF once and use different versioning prefix in the view
+    final Collection<Function> functions = registry.lookup(removeVersioningPrefix(funcClassName));
+    if (functions.isEmpty()) {
       Collection<Function> dynamicResolvedFunctions =
           resolveDaliFunctionDynamically(functionName, funcClassName, hiveTable, numOfOperands);
 
-      if (dynamicResolvedFunctions.size() == 0) {
+      if (dynamicResolvedFunctions.isEmpty()) {
         // we want to see class name in the exception message for coverage testing
         // so throw exception here
         throw new UnknownSqlFunctionException(funcClassName);
       }
 
       return dynamicResolvedFunctions;
+    } else {
+      if (isVersioningUDF(funcClassName)) {
```

Comment:
The logic here should be:
* If it has the necessary properties (and hence Dali UDF)
  * If it is on the format `[name]+_version` or `db_t_+[name]`, then
    *  `short name = [name]`.
    * `function name = unversioned class name`
    * `long name = [name]+_version or db_t_+[name] or whatever original name is`
  * else a special case of the above
  
Ideally Dali views have `function-identifier` property that makes it possible not derive function name from the unversioned class name but the latter is acceptable in the interim, especially everything is encapsulated in this method.

Reply:
Also in case `short name = [name]` is not useful for Trino, we can have `short name` depend on `function name` through a hardcoded dependency. Intention of `short name` is to be used in engines that cannot leverage versioning like Trino.

Reply:
Modified the code by adding a new method `getTrinoFunctionName` to `VersionedSqlUserDefinedFunction`, which returns the Trino function name as how we did before:
1. Try to retrieve the Trino function name from the map `TRINO_FUNC_NAME_MAP` based on the UDF class name
2. If the UDF class name doesn't exist in the map, follow the previous convention by converting the camel case class name to lowercase underscore function name.

Since we don't have a standard for version in view function name (we only have a standard for versioning prefix in class name), and view function name may not be related to the registered Trino function name, I would prefer to keep the current approach of generating Trino function name.